h/views/api/helpers/links.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+"""Helpers for generating and retrieving service link metadata"""
+
+from __future__ import unicode_literals
+
+
+class ServiceLink(object):
+    """Encapsulate metadata about an API service"""
+
+    def __init__(self, name, route_name, method="GET", description=None):
+        self.name = name
+        self.route_name = route_name
+        self.method = method
+        self.description = description
+
+    def primary_method(self):
+        """
+        Determine the primary HTTP method for this service
+
+        If the ``method`` indicated is a tuple, the first entry will be
+        returned.
+
+        :rtype: str
+        """
+        method = self.method
+        if isinstance(method, tuple):
+            # If the view matches multiple methods, assume the first one is
+            # preferred
+            method = method[0]
+        return method
+
+    def asdict(self):
+        """
+        Return a ``dict`` representation of the link
+
+        :rtype: dict
+        """
+        return {
+            "name": self.name,
+            "description": self.description,
+            "method": self.primary_method(),
+            "route_name": self.route_name,
+        }
+
+
+def register_link(link, versions, registry):
+    """
+    Register an API service's metadata for its supported versions
+
+    Add an entry for the indicated ``link`` onto the ``registry`` for each
+    of the versions it claims to supported. These entries include basic
+    endpoint metadata like description and primary HTTP method.
+
+    These data structures are later referenced to deliver an index of our
+    API endpoints at `GET /api/`.
```

Comment:
Single backticks render as italics in docstring HTML, so this is actually intentional. Neener!

Reply:
I think they actually render as `<cite>...</cite>`, which the CSS styles as italics. You can configure Sphinx to make single backticks do `<code>...</code>` as they do in Markdown, we should configure that